workshops/views.py

```diff
@@ -3136,67 +3260,302 @@ def trainingrequest_create(request):
+            # Perform bulk discard
+            for r in form.cleaned_data['requests']:
+                r.state = 'd'
+                r.save()
 
-@login_required
-def autoupdate_profile(request):
-    person = request.user
-    form = AutoUpdateProfileForm(instance=person)
+            messages.success(request, 'Successfully bulk discarded selected '
+                                      'requests.')
 
-    if request.method == 'POST':
-        form = AutoUpdateProfileForm(request.POST, instance=person)
+            return redirect(request.get_raw_uri())
 
-        if form.is_valid() and form.instance == person:
-            # save lessons
-            person.lessons.clear()
-            for lesson in form.cleaned_data['lessons']:
-                q = Qualification(lesson=lesson, person=person)
-                q.save()
+    elif request.method == 'POST' and 'unmatch' in request.POST:
+        # Bulk unmatch people associated with selected TrainingRequests from
+        # trainings.
+        form = BulkChangeTrainingRequestForm(request.POST)
+        match_form = BulkMatchTrainingRequestForm()
 
-            # don't save related lessons
-            del form.cleaned_data['lessons']
+        form.check_person_matched = True
+        if form.is_valid():
+            # Perform bulk unmatch
+            for r in form.cleaned_data['requests']:
+                r.person.training_tasks.delete()
 
-            person = form.save()
+            messages.success(request, 'Successfully bulk unmatched selected '
+                                      'people from trainings.')
+
+            return redirect(request.get_raw_uri())
+
+    else:  # GET request
+        form = BulkChangeTrainingRequestForm()
+        match_form = BulkMatchTrainingRequestForm()
+
+    context = {'title': 'Training Requests',
+               'requests': requests,
+               'filter': filter,
+               'form': form,
+               'match_form': match_form}
+
+    return render(request, 'workshops/all_trainingrequests.html', context)
+
+
+def _accept_training_request(form, create_new_person, training_request,
+                             request):
+    try:
+        if create_new_person:
+            training_request.person = Person.objects.create_user(
+                username=create_username(training_request.personal,
+                                         training_request.family),
+                personal=training_request.personal,
+                family=training_request.family,
+                email=training_request.email,
+            )
+
+    except IntegrityError as e:
+        # FIXME Any idea how to narrow scope of this except clause only to
+        # IntegrityErrors related to email address?
+
+        # email address is not unique
+        messages.error(request, 'Could not create a new person -- '
+                                'there already exists a person with '
+                                'exact email address.')
+        return None
```

Comment:
Control flow. We want to exit `trainingrequest_details` view with returned redirection. If it's `None`, we continue execution of this view.